### PR TITLE
Archer: Bug Fix: logout-zero-records-fix

### PR DESCRIPTION
**Pull request by Trek333**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [**no**] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [**no**] If this is a new connector or you are adding new actions
    - [na ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [na] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [**no**] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [**no**] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [**no**] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
Bug fixes to logout of RSA Archer API at end of actions and return success for 0 records of 'list tickets' action

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by achandya-crest** [comment]
Hi @Trek333,
While testing your submission of the RSA Archer app we found an issue with the update_ticket action. When not providing the content ID, name field, or name value, ideally it should return an error message stating "Either content ID or both name field and name value are mandatory". However, it is showing an error message as "Either content ID or both name field and name value are mandatory - Terminate session details: 'NoneType' object has no attribute 'terminate_session'". With app_version 2.1.6 this action is running fine with the proper error message. Hence, we request you to please look into this issue and provide us your inputs if you were trying to get such output only because we assume terminate_session should not be done here?

**Comment by Trek333** [comment]
Hi @achandya-crest 
Nice catch! The _handle_update_ticket method was updated to resolve the issue that you found. Please review and let me know your feedback.

File: release_notes/2.1.7.md [hunk 1]
```diff
@@ -0,0 +1 @@
+* Bug fixes to logout of RSA Archer API at end of actions and return success for 0 records of 'list tickets' action
```
Comment (dvora-crest) [inline release_notes/2.1.7.md:1]: This file can be removed as the pipeline automatically creates a version-specific md file for release notes from the unreleased.md file on merge.

File: release_notes/unreleased.md [hunk 2]
```diff
@@ -1 +1,2 @@
 **Unreleased**
+* Bug fixes to logout of RSA Archer API at end of actions and return success for 0 records of 'list tickets' action
```
Comment (dvora-crest) [inline release_notes/unreleased.md:2]: As multiple bugs are fixed, can we mention each fix separately?
* Bug fix to logout of RSA Archer API at the end of actions
* Bug fix to return success for 0 records of 'list tickets' action

File: release_notes/2.1.7.md (hunk 1 above)
Comment (achandya-crest) [reply release_notes/2.1.7.md]: Done

File: release_notes/unreleased.md (hunk 2 above)
Comment (achandya-crest) [reply release_notes/unreleased.md]: Done

**Review by dhwanis-crest — APPROVED** [review]

**Review by dvora-crest — APPROVED** [review]

**Pull request merged**